crypto/fipsmodule/sha/keccak1600.c

```diff
@@ -0,0 +1,1112 @@
+/*
+ * Copyright 2016 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <assert.h>
+#include <string.h>
+
+
+#if !defined(KECCAK1600_ASM)
+ // Default to KECCAK_2X variant
+#if !defined(KECCAK_REF) && !defined(KECCAK_1X) && !defined(KECCAK_1X_ALT) && \
+    !defined(KECCAK_2X) && !defined(KECCAK_INPLACE)
```

Comment:
this file contains many Keccak implementations. Do we need all of them and is the code complexity worth it (especially if we are planning on importing assembly as well)?

Reply:
I agree that they are not needed since we will add different target-specific assembly implementations. 
I kept only Keccak_x2 since it makes use of large register banks for data caching (it is most suitable for high-end devices which is the main target group of AWS-LC); it is also the default OpenSSL implementation design. I can add others later or replace with another design if needed.